webclient/app/src/app/commons/gitDownloadButton/GitDataButton.jsx

```diff
@@ -0,0 +1,144 @@
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import { useTranslation } from 'react-i18next';
+import { CloudSync, Close } from "@mui/icons-material";
+import ErrorAlert from "../alert/ErrorAlert";
+import NotificationDialog from "../alert/NotificationDialog";
+import SimpleValidatedTextField from "../validatedTextField/SimpleValidatedTextField";
+import RegexConfig from "../../../regexConfig";
+import GitDataButtonStyles from "./GitDataButtonStyles";
+
+function GitDataButton(props) {
+    const { credentialsRequired, handleGit, handleAfterSuccess, buttonIcon, buttonName, buttonVariant } = props;
+    const gitDataButtonStyles = GitDataButtonStyles();
+    const [hasFormError, setHasFormError] = React.useState(false);    
+    const [downloadRequestData, setDownloadRequestData] = useState({ "username": "", "password": "" });
+    const [alert, setAlert] = useState({"open":false, "title": "ERROR", "message": ""});
+    const { t } = useTranslation();
+
+    const [openAuthDialog, setOpenAuthDialog] = useState(false);
+  
+
+    const handleChange = (event) => {
+        const { name, value } = event.target;
+        let downloadRequestDataNew = { ...downloadRequestData };
+        downloadRequestDataNew[name] = value;
+        setDownloadRequestData(downloadRequestDataNew);
+    }
+
+    const handleLogin = () => {
+        if (credentialsRequired) {
+            handleAlertClose();
+            setOpenAuthDialog(true);
+        } else {
+            handleAppTemplateReload();
+        }
+    }
+
+    const handleAppTemplateReload = () => {
+        if(hasFormError && credentialsRequired) {
+            return;
+        }
+        handleGit(downloadRequestData).then(() => {
+            handleAfterSuccess();
+            setOpenAuthDialog(false);
+        }).catch(err => {
+            setAlert({"open":true, "title": t("alert.error"), "message": t(err.response.data.messageKey)});
+            console.log(err.response.data);
+        });
+    }
+
+    const handleAlertClose = () => {
+        setAlert({"open":false, "title": "ERROR", "message": ""});
+    }
+
+    const onClose = () => {
+        setDownloadRequestData({ "username": "", "password": "" });
+        setOpenAuthDialog(false);
+    }
+
+    useEffect(() => {
+        if (!downloadRequestData) {
+            return;
+        }
+        let hasError = false;
+
+        if (!RegexConfig.appTemplateAuthUser.test(downloadRequestData.username)) {
+            hasError = true;
+        }
+        if (!RegexConfig.appTemplateAuthPassword.test(downloadRequestData.password)) {
+            hasError = true;
+        }
+        setHasFormError(hasError);
+    }, [downloadRequestData, hasFormError])
+
+
+    return (
+        <Container>
+            <Button onClick={handleLogin} startIcon={buttonIcon ? buttonIcon : <CloudSync />} variant={buttonVariant} >{buttonName}</Button>
```

Comment:
why do you need the evaluation of buttonIcon?
```suggestion
            <Button onClick={handleLogin} startIcon={<CloudSync />} variant={buttonVariant} >{buttonName}</Button>
```

Reply:
because we have a git clone for app and one for appTemplate. the app is downloaded and apptemplate is being refreshed in the database and UI.